api/envoy/extensions/clusters/dynamic_forward_proxy/v3/cluster.proto

```diff
@@ -54,3 +75,29 @@ message ClusterConfig {
   //
   bool allow_coalesced_connections = 3;
 }
+
+// [#next-free-field: 6]
+message SubClustersConfig {
+  // The :ref:`service discovery type <arch_overview_service_discovery_types>`
+  // to use for resolving a sub cluster. Hard code STRICT_DNS now.
+  // And we may also support LOGICAL_DNS, when we deprecate the dns_cache_config configuration.
+  //
+  // [#not-implemented-hide:]
+  config.cluster.v3.Cluster.DiscoveryType type = 1 [(validate.rules).enum = {defined_only: true}];
```

Comment:
Just remove this completely and document that the sub-cluster is hard coded to STRICT_DNS.